From 3ac28dca1710d12dfe290e264c3f57cca0e5e373 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bernhard=20=C3=9Cbelacker?= <bernhardu@mailbox.org>
Date: Thu, 19 Dec 2024 16:46:09 -0800
Subject: [PATCH 02/15] ntdll: Add read and write to commited guard pages.

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=55138

wine bash.exe
notepad.exe
wine: Unhandled page fault on write access to 00000007FFFF81F8 at address 0000000180265986 (thread 0130), starting debugger...

wine bash.exe -c 'WIN_ROOT="$(PATH=${MSYS2_PATH} exec /usr/bin/cygpath -Wu)"; echo $WINROOT'
/usr/bin/bash: Zeile 1: /usr/bin/cygpath: Bad address
/usr/bin/bash: Zeile 1: /usr/bin/cygpath: No error

Based on: https://bugs.winehq.org/attachment.cgi?id=70483
  (Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=47808)
---
 dlls/ntdll/unix/virtual.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index 3629fd1b2c..3a2bbcd0da 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -4559,7 +4559,7 @@ static NTSTATUS grow_thread_stack( char *page, struct thread_stack_info *stack_i
     mprotect_range( page, host_page_size, 0, 0 );
     if (page >= stack_info->start + host_page_size + stack_info->guaranteed)
     {
-        set_page_vprot_bits( page - host_page_size, host_page_size, VPROT_COMMITTED | VPROT_GUARD, 0 );
+        set_page_vprot_bits( page - host_page_size, host_page_size, VPROT_COMMITTED | VPROT_READ | VPROT_WRITE | VPROT_GUARD, 0 );
         mprotect_range( page - host_page_size, host_page_size, 0, 0 );
     }
     else  /* inside guaranteed space -> overflow exception */
-- 
2.53.0

